CMakeLists.txt

```diff
@@ -49,12 +57,25 @@ include_directories(extern/rapidjson/include)
 
 add_subdirectory(src)
 
-find_package(Boost 1.50 REQUIRED COMPONENTS
-	program_options
-)
+# Tests
+file(GLOB TEST_SRCS tests/cpp/*.cpp)
+find_library(GTEST_LIB gtest "$ENV{GTEST_LIB}")
 
-add_executable(${PROJECT_NAME}_test src/test.cpp)
-target_link_libraries(${PROJECT_NAME}_test PUBLIC ${PROJECT_LIB_NAME} ${Boost_LIBRARIES})
+if(GTEST_LIB)
```

Comment:
I suggest creating a CMakeLists.txt in the tests folder and move codes inside if clause.